### PR TITLE
Resource-based functions

### PR DESCRIPTION
**Pull request by xgrg**
Moved specific resource-based functions from `master` to separate branch.

### CODE REVIEW

File: pyxnat/core/derivatives/validator.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+XNAT_RESOURCE_NAMES = ['BBRC_VALIDATOR']
+
+def tests(self, name, key=None):
+    import json
+    j = [e for e in list(self.files(name)) if 'json' in e._urn][0]
```
Comment (jhuguetn) [inline pyxnat/core/derivatives/validator.py]:
```suggestion
    j = [e for e in list(self.files('{}*.json'.format(name)))][0]
```

Comment (jhuguetn) [reply pyxnat/core/derivatives/validator.py]: I find it hard for end-user to provide full matching file name (which is composed validator name and MRSession label)

File: pyxnat/core/pipelines.py [hunk 2]
```diff
@@ -2,14 +2,20 @@
...
-import urllib2
-import smtplib
-from copy import deepcopy
-import email.mime.text
-import xml.dom.minidom
 
-import suds.client
-import suds.xsd.doctor
+import six
+if six.PY2:
```
Comment (jhuguetn) [inline pyxnat/core/pipelines.py]: I don't think `urlopen` is ever used

Comment (xgrg) [reply pyxnat/core/pipelines.py]: Actually it is used further down in the commented part. The previous import urlopen was failing due to Python 3 incompatibility. Shall we just comment the whole import?

Comment (jhuguetn) [reply pyxnat/core/pipelines.py]: Agree

**Pull request merged**